sympy/vector/lamecoeff.py

```diff
@@ -0,0 +1,77 @@
+from sympy.core.basic import Basic
+from sympy import sqrt, sin, cos
+from sympy.core.function import Derivative
+
+
+class CoeffProvider(Basic):
+    """
+    It provides specific information of curvilinear coordinate
+    system to CoordSysCartesian.
+
+    Ideally, users should not instantiate this class.
+
+    """
+
+    def __new__(cls, system, curv_coord_name=None, transformation_equations=None):
```

Comment:
I meant to suggest a single argument, use the same argument for all cases. SymPy has an expression rewriting engine that requires us to write the constructors in a particular way.